### ISSUE TITLE
Persistent headers broken with subpages

### ISSUE THREAD

**Issue by Vinky** [ToddThomson/jQuery-Mobile-Subpage-Widget#11]
Hi,

JQM 1.1.0 introduced fixed headers / footers that work pretty well, and it would be awesome if your plugin could handle it.
I didn't dwell into JQM and your code (and I apologize), but set up two Fiddles to show you the issue :
- How it should work (i.e without subpages) : http://jsfiddle.net/Vinky/YJBQR/
- How it actually works (with subpages) : http://jsfiddle.net/Vinky/LMB2y/

I think it could be a real improvement to your plugin.
Cheers,

Vinky

**Comment by ToddThomson** [comment]
Thank-you for this.
I noticed that in jQM 1.1 that the parent page gets assigned the class ui-page-header-fixed when it contains a header which is fixed. This sets the top padding on the page so that content is shifted down below the bottom of the fixed header.
I will put this issue on my fix list for the next version.
You should be able to avoid/workaround this issue by adding a css rule to the class ui-subpage when it has fixed headers ( and footers ).

**Comment by Vinky** [comment]
Thanks for your reply, but my concern was not so much about the padding (that can indeed easily be corrected), but rather on the persistent aspect of the header. When the transition is kicked, the header should remain in place while the page slides.

**Comment by ToddThomson** [comment]
I looked at my development code for my jQM 1.1 based app to see why I was not running into this issue. jQM ALSO dynamically sets the top padding on the parent page. With my app the top padding get set properly so that the page content gets shifted down.

**Comment by ToddThomson** [comment]
Yes, I also see the difference with the persistent header. I did not realize that if the header data-id was set to the same value in 2 pages that the header would not also transition when navigating between the two. Is this really a documented behaviour in jQM ( If the 2 data-ids are different the header does transition ). Can you give me a use for transitioning the page content while keeping the header the same ( and in place )?
When I fix the padding issue, I will take a look at the transition issue as well.

**Comment by Vinky** [comment]
jQM's documentation on persistent header / footer :
http://jquerymobile.com/demos/1.1.0/docs/toolbars/footer-persist-d.html

I personally use this behaviour for pagination ; the header displays the application name, while the content displays several pages, that have the same context. This gives a web application a more native feel.

**Comment by ToddThomson** [comment]
Persistent footer toolbars must have the persistent/in-place behaviour. It never occurred to me that anyone would want fixed (non-transitioning) headers across page transitions! However, it is not up to me to say what is right or wrong!
I will add this as a separate issue to be fixed in 1.1

**Comment by ToddThomson** [comment]
@Vinky I've looked into this issue and have found the root cause. The problem is that in jQM 1.1 fixed toolbars are processed/enhanced within the subpage before the subpage is enhanced and thus the "ui-page-header-class" is applied to the parent page ( of the subpages ). In my opinion this is yet another issue with the fixed toolbar widget's poor design and implementation.
I have added an issue to jQM: #4496.
There seems to be several ways to work around this issue, but the easiest thing would be to add the functionality to pageLoad() in jQM 1.1.1!
Having the subpage widget do it's work before the parent page is processed would also be an acceptable solution in the short term ( bind on the "beforecreate" ). I will investigate this now

**Comment by Vinky** [comment]
jQM's handling of fixed headers is quite hard to follow, indeed.
Thanks for your concern, I'm looking forward to your investigations' results :)

**Comment by ToddThomson** [comment]
@Vinky I have updated the subpage widget to fix this issue.

**Comment by Vinky** [comment]
I thought I tried binding the widget's initialization to pagebeforeshow, but your fix solves the persistent behaviour, so I guess I did it wrong, or not at all :)
Thanks, time to close the issue.
